### PR TITLE
Add format output flag for versionCmd

### PR DESCRIPTION
**Pull request by Frankonly**
I take the version command as an example of format output refactoring. #935 
In new versionCmd, by default, result is printed through stdout and error is printed throught stderr.
When using the flag output-format(eg. `-o json`), output(contains error) will be printed through stdout.

Tested on windows and centos

```
// success case
-> ioctl version -o json                                      
{
  "messageType": 0,
  "message": {
    "object": "Client",
    "versionInfo": {
      "packageVersion": "v0.8.0-13-gfa27c17",
      "packageCommitID": "fa27c17a0c286d6df5f3c35ad9ae2c2b38e450a7",
      "gitStatus": "clean",
      "goVersion": "go version go1.12.6 linux/amd64",
      "buildTime": "2019-07-08-CST/13:12:02"
    }
  }
}
{
  "messageType": 0,
  "message": {
    "object": "api.iotex.one:443",
    "versionInfo": {
      "packageVersion": "v0.6.2",
      "packageCommitID": "2df78e9460dee3bd3d96526468472409ac36d615",
      "gitStatus": "clean",
      "goVersion": "go version go1.12.5 linux/amd64",
      "buildTime": "2019-05-29-UTC/21:33:14"
    }
  }
}


// failure case
-> ioctl version -o json
{
  "messageType": 0,
  "message": {
    "object": "Client",
    "versionInfo": {
      "packageVersion": "v0.8.0-13-gfa27c17",
      "packageCommitID": "fa27c17a0c286d6df5f3c35ad9ae2c2b38e450a7",
      "gitStatus": "clean",
      "goVersion": "go version go1.12.6 linux/amd64",
      "buildTime": "2019-07-08-CST/13:12:02"
    }
  }
}
{
  "messageType": 3,
  "message": {
    "code": 1,
    "info": "all SubConns are in TransientFailure, latest connection error: conn                                                                                        ection error: desc = \"transport: Error while dialing dial tcp: address baidu.co                                                                                        m: missing port in address\""
  }
}

```

### CODE REVIEW

File: ioctl/output/format.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+)
+
+// Output is used for format output
+type Output struct {
+	Error   Error   `json:"error"`
+	Message Message `json:"message"`
+}
+
+// Error is the error part of Output
+type Error struct {
```
Comment (yutongp) [inline ioctl/output/format.go]: it not common an `error` struct is not a go error

File: ioctl/cmd/version/version.go [hunk 2]
```diff
@@ -63,9 +81,37 @@ func version() (string, error) {
...
+		return oErr
+	}
+
+	message.VersionInfo = response.ServerMeta
+	printVersion(message, emptyError)
+	return oErr
+}
+
+func printVersion(message versionMessage, oErr output.Error) {
+	switch {
```
Comment (yutongp) [inline ioctl/cmd/version/version.go]: just switch onn `output.OutputFormat`?

Comment (Frankonly) [reply ioctl/cmd/version/version.go]: yes

Comment (Frankonly) [reply ioctl/cmd/version/version.go]: > just switch onn `output.OutputFormat`?

yes

File: ioctl/cmd/version/version.go [hunk 3]
```diff
@@ -63,9 +73,27 @@ func version() (string, error) {
...
+			"failed to get version from server: "+err.Error())
+	}
+
+	message.VersionInfo = response.ServerMeta
+	printVersion(message)
+	return nil
+}
+
+func printVersion(message versionMessage) {
+	switch {
```
Comment (zjshen14) [inline ioctl/cmd/version/version.go]:
```suggestion
	switch output.OutputFormat {
```

File: ioctl/output/format.go [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+)
+
+// Output is used for format output
+type Output struct {
+	MessageType MessageType `json:"messageType"`
+	Message     Message     `json:"message"`
+}
+
+// Message is the message part of output
+type Message interface {
```
Comment (zjshen14) [inline ioctl/output/format.go]: Is this interface necessary?

File: ioctl/output/format.go [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+// ErrorMessage is the struct of an Error output
+type ErrorMessage struct {
+	Code ErrorCode `json:"code"`
+	Info string    `json:"info"`
+}
+
+// PrintError prints error message in format, and returns golang error when using default output
+func PrintError(code ErrorCode, info string) error {
+	switch {
```
Comment (zjshen14) [inline ioctl/output/format.go]: Ditto

**Pull request merged**